### PR TITLE
Don't start byebug after disable-pry

### PR DESCRIPTION
**Pull request by printercu**
Following https://github.com/deivid-rodriguez/pry-byebug/pull/102#issuecomment-262350015

### CODE REVIEW

**Comment by printercu** [comment]
ping @deivid-rodriguez 😃

**Comment by deivid-rodriguez** [comment]
Hei! Thanks (again)! :)

This sounds sensible. Do you think you could add a test for it? Maybe "translating" what's in `pry`'s wiki into a test: https://github.com/pry/pry/wiki/FAQ#how-do-i-stop-executing-a-bindingpry-call-in-a-loop? If you can't I'll still merge this, it'll just take more time.

**Comment by printercu** [comment]
@deivid-rodriguez seems like it should be tested with mock library, because what we really want to test is that `Byebug::PryProcessor` doesn't receive `start` after `disable-pry`. Should I add mocha for this? I'm not familiar with minitest, but I've found only `assert_send` which can not be used for this.

**Comment by deivid-rodriguez** [comment]
Why not an integration test like the one I'm suggesting?

**Comment by printercu** [comment]
With minitest? Don't really understand, how integration test would check that byebug is not started after `disable-pry`. Or what kind of test do you mind?

**Comment by deivid-rodriguez** [comment]
Have a look at one of the current tests: https://github.com/deivid-rodriguez/pry-byebug/blob/master/test/commands/frames_test.rb#L18-L29

You specify a list of inputs to be fed to `pry`, an example program to be run and the expected output pry-byebug should give. Makes sense for this case?

**Comment by deivid-rodriguez** [comment]
Oh, wait, soory I think I'm misunderstanding this.

It's not that `disable-pry` doesn't work with `pry-byebug`, it's just that it slows down the debugger because it starts `byebug`. Right?

In that case, yes, you're right that an integration test like the one I'm suggesting doesn't help.

**Comment by printercu** [comment]
I've got this:

```ruby
class PryExtTest < MiniTest::Spec
  let(:output) { StringIO.new }
  let(:input) { InputTester.new }

  describe 'when disable-pry called' do
    before { input.add 'disable-pry' }
    after { ENV.delete('DISABLE_PRY') }

    it 'shows current line' do
      # This line is analog for rspec:
      expect(Byebug::PryProcessor).to receive(:start).and_call_original
      redirect_pry_io(input, output) { load test_file('multiple') }
    end
  end
end

# multiple.rb
5.times do
  binding.pry
end
```

pry-byebug works fine with `disable-pry` (subsequent calls to binding.pry are ignored), except it starts byebug  and everything gets slow. So I don't see any way to test it, except checking that `.start` is called only once. And I don't see how to implement it similar to your example :(

**Comment by printercu** [comment]
Yep 😄

**Comment by printercu** [comment]
Ah. I got it. Maybe I can check that byebug is not running right after example. Just a moment, I'll prepare commit.

**Comment by deivid-rodriguez** [comment]
Great! :)

**Comment by printercu** [comment]
Ready

**Comment by deivid-rodriguez** [comment]
Sweeeeeet! :heart: :yellow_heart: :purple_heart:

Just a nitpick. Can we split the test into two? One checking that byebug is not started (the `Byebug.stop/started` dance) and other checking that the default `pry` behaviour is kept (the output checking dance).

Oh, and could you add a Change log entry?

**Comment by printercu** [comment]
Done

**Comment by deivid-rodriguez** [comment]
Thanks! Mmmm, the build is failing now... :S

**Comment by printercu** [comment]
Yep. I haven't seen it on my computer, because other test modify pry's config. I've moved this initialization to `test_helper.rb`.

**Comment by deivid-rodriguez** [comment]
@printercu Sorry I didn't reply again, I noticed that the build was segfaulting against `ruby-head` and was doing some investigations. Result (as expected): not this patch's fault at all.

Just two last nitpicks and I'm merging:

* Could you change the first test description to `"keeps pry's default behaviour"`? Having a backquote inside regular text feels very weird to me when reading it.

* The second test description is a bit misleading, no? Shouldn't it be "does not start byebug" instead of "stops byebug"?

**Comment by printercu** [comment]
ready

File: test/pry_ext_test.rb [hunk 1]
```diff
@@ -1,4 +1,23 @@
...
+    subject { redirect_pry_io(input, output) { load test_file('multiple') } }
+    before { input.add 'disable-pry' }
+    after { ENV.delete('DISABLE_PRY') }
+
+    it "keeps pry's default behaviour" do
+      subject
+      assert_equal 1, output.string.scan(/binding\.pry/).size
+    end
+
+    it 'does not strt byebug' do
```
Comment (deivid-rodriguez) [inline test/pry_ext_test.rb]: Typo :bowtie:

Comment (printercu) [inline test/pry_ext_test.rb]: oops. fixed

**Comment by deivid-rodriguez** [comment]
Thanks @printercu! :)

**Comment by printercu** [comment]
Oh, cool!

**Comment by deivid-rodriguez** [comment]
@printercu Just released 3.4.2 with your fix :wink:

**Pull request merged**